oidc_auth/authentication.py

```diff
@@ -101,71 +139,77 @@ def get_jwt_value(self, request):
         auth = get_authorization_header(request).split()
         auth_header_prefix = api_settings.JWT_AUTH_HEADER_PREFIX.lower()
 
-        if not auth or smart_text(auth[0].lower()) != auth_header_prefix:
+        if not auth or smart_str(auth[0].lower()) != auth_header_prefix:
             return None
 
         if len(auth) == 1:
             msg = _('Invalid Authorization header. No credentials provided')
             raise AuthenticationFailed(msg)
         elif len(auth) > 2:
-            msg = _('Invalid Authorization header. Credentials string should not contain spaces.')
+            msg = _(
+                'Invalid Authorization header. Credentials string should not contain spaces.')
             raise AuthenticationFailed(msg)
 
         return auth[1]
 
+    # @cache(ttl=api_settings.OIDC_JWKS_EXPIRATION_TIME)
     def jwks(self):
-        keys = KEYS()
-        keys.load_jwks(self.jwks_data())
-        return keys
-
-    @cache(ttl=api_settings.OIDC_JWKS_EXPIRATION_TIME)
-    def jwks_data(self):
         r = request("GET", self.oidc_config['jwks_uri'], allow_redirects=True)
         r.raise_for_status()
-        return r.text
+        keys = JsonWebKey.import_key_set(r.json())
+        return keys
 
     @cached_property
     def issuer(self):
         return self.oidc_config['issuer']
 
-    def decode_jwt(self, jwt_value):
-        keys = self.jwks()
+    def decode_jwt(self, jwt_value: bytes):
         try:
-            id_token = JWS().verify_compact(jwt_value, keys=keys)
-        except (JWKESTException, ValueError):
-            msg = _('Invalid Authorization header. JWT Signature verification failed.')
+            # assert isinstance(jwt_value, bytes)
+            claims_options = {
+                'iss': {
+                    'essential': True,
+                    'value': self.issuer
+                },
+                'aud': {
+                    'values': self.get_audiences()
+                }
+            }
```

Comment:
This would perhaps be nice as a constant at the top of the file? :)

Reply:
I made it as a property.